### ISSUE TITLE
[SCM-836] scm:bootstrap throws NPE

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-scm#1259]
**[Dan Tran](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dantran)** opened **[SCM-836](https://issues.apache.org/jira/browse/SCM-836?redirect=false)** and commented

This issue shows up at 1.9.5 with the following stack trace

Caused by: java.lang.NullPointerException
at org.apache.maven.scm.plugin.BootstrapMojo.runGoals(BootstrapMojo.java:123)
at org.apache.maven.scm.plugin.BootstrapMojo.execute(BootstrapMojo.java:94)
at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
... 21 more

Looking into debug log. Looks like mavenHome is set to null which should be "" instead ( ie defaultValue="" does not work??? )

[DEBUG] Configuring mojo 'org.apache.maven.plugins:maven-scm-plugin:1.9.5:bootstrap' with basic configurator -->
[DEBUG]   (f) basedir = /data/jenkins-work/jobs/sit/jobs/sit-master/workspace/sit/aaa
[DEBUG]   (s) checkoutDirectory = /data/jenkins-work/jobs/sit/jobs/sit-master/workspace/sit/aaa/target/checkout
[DEBUG]   (f) connectionType = connection
[DEBUG]   (f) developerConnectionUrl = scm:p4://IDP/shared/main/itests/restapi/aaa
[DEBUG]   (f) goals = verify,-Dsky.sshUser=admin
[DEBUG]   (f) profiles = aaa
[DEBUG]   (f) pushChanges = true
[DEBUG]   (f) settings = org.apache.maven.execution.SettingsAdapter@5b69d40d
[DEBUG]   (f) skipCheckoutIfExists = false
[DEBUG]   (f) useExport = false
[DEBUG]   - end configuration --

work around is to pass in -DmavenHome=""

---

**Affects:** 1.9.5

1 votes, 2 watchers

**Comment by jira-importer** [comment]
**[Dan Tran](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dantran)** commented

this is to confirm that

+ /**

+ 

      * The path where you maven is installed

+ 

      */

+ `@Parameter`( property = "mavenHome", defaultValue = "")

+ private String mavenHome;

is set to null

quicktest with mvn 3.3.9

**Comment by jira-importer** [comment]
**[Dan Tran](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dantran)** commented

also mavenHome should default to the same started maven via

```
String mvnPath = System.getProperty("maven.home") + "/bin/mvn";
if (Os.isFamily("windows")) {
    String winMvnPath = mvnPath + ".bat";
    if (!new File(winMvnPath).exists()) {
        winMvnPath = mvnPath + ".cmd";
    }
    mvnPath = winMvnPath;
}
```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [0be4b4259af486bf8cbf218a30be3af2c12e910f](https://gitbox.apache.org/repos/asf?p=maven-scm.git;a=commit;h=0be4b4259af486bf8cbf218a30be3af2c12e910f).
